ui/appui-react/src/appui-react/statusbar/StatusBarComposer.tsx

```diff
@@ -355,12 +386,18 @@ export function StatusBarComposer(props: StatusBarComposerProps) {
   const handleEntryResize = React.useCallback(
     (key: string) => (w: number) => {
       const oldW = entryWidths.current.get(key);
-      if (undefined === oldW || Math.abs(oldW - w) > 2) {
+      if (oldW === undefined || Math.abs(oldW - w) > 2) {
         entryWidths.current.set(key, w);
+        if (oldW === undefined) {
+          const widths = verifiedMapEntries(entryWidths.current);
+          if (widths !== undefined)
+            entryWidths.current = sortWidths(widths, statusBarItems);
```

Comment:
@GerardasB could you take a look at `fix overflow order` commit? I managed to reproduce the issue of items going to overflow in the wrong order. I reproduced it by adding a new item provider after a timeout. My fix is basically sorting the widths every time a new status bar item is added. It is not a clean solution but this is the best I could do since the code is very interconnected and old (and needs a refactoring but I don't want to do that right now because this file is >500 lines). I tried a few different ways to fix this issue e.g. resetting `entryWidths`, sorting in a `useMemo` but none of them worked that well.

Reply:
Seems good and might be worth mentioning in the `NextVersion`.
As discussed - long term we should have a single place where we handle overflow logic, since we have multiple components that need it.